tests/testthat/test-group.R

```diff
@@ -2,120 +2,138 @@ context('group')
-  dt <- groupSummary(data.xy, NULL, 'y')
+  dt <- groupSummary(data.xy, NULL, 'entity.y')
   expect_is(dt, 'data.table')
   expect_equal(nrow(dt),1)
   expect_equal(names(dt), c('min', 'q1', 'median', 'q3', 'max'))
  
-  dt <- groupSummary(data.xy, NULL, 'y', 'group')
+  dt <- groupSummary(data.xy, NULL, 'entity.y', 'entity.group')
   expect_is(dt, 'data.table')
   expect_equal(nrow(dt),4)
-  expect_equal(names(dt), c('group', 'min', 'q1', 'median', 'q3', 'max'))
+  expect_equal(names(dt), c('entity.group', 'min', 'q1', 'median', 'q3', 'max'))
 
-  dt <- groupSummary(data.xy, NULL, 'y', NULL, 'panel')
+  dt <- groupSummary(data.xy, NULL, 'entity.y', NULL, 'entity.panel')
   expect_is(dt, 'data.table')
   expect_equal(nrow(dt),4)
-  expect_equal(names(dt), c('panel', 'min', 'q1', 'median', 'q3', 'max'))
+  expect_equal(names(dt), c('entity.panel', 'min', 'q1', 'median', 'q3', 'max'))
 
-  dt <- groupSummary(data.xy, NULL, 'y', 'group', 'panel')
+  dt <- groupSummary(data.xy, NULL, 'entity.y', 'entity.group', 'entity.panel')
   expect_is(dt, 'data.table')
   expect_equal(nrow(dt),16)
-  expect_equal(names(dt), c('group', 'panel', 'min', 'q1', 'median', 'q3', 'max'))
+  expect_equal(names(dt), c('entity.group', 'entity.panel', 'min', 'q1', 'median', 'q3', 'max'))
 
-  dt <- groupSummary(data.catXcontY, 'x', 'y')
+  dt <- groupSummary(data.catXcontY, 'entity.x', 'entity.y')
   expect_is(dt, 'data.table')
   expect_equal(nrow(dt),1)
-  expect_equal(names(dt), c('x', 'min', 'q1', 'median', 'q3', 'max'))
+  expect_equal(names(dt), c('entity.x', 'min', 'q1', 'median', 'q3', 'max'))
 
-  dt <- groupSummary(data.catXcontY, 'x', 'y', 'group')
+  dt <- groupSummary(data.catXcontY, 'entity.x', 'entity.y', 'entity.group')
   expect_is(dt, 'data.table')
   expect_equal(nrow(dt),4)
-  expect_equal(names(dt), c('group', 'x', 'min', 'q1', 'median', 'q3', 'max'))
+  expect_equal(names(dt), c('entity.group', 'entity.x', 'min', 'q1', 'median', 'q3', 'max'))
 
-  dt <- groupSummary(data.catXcontY, 'x', 'y', NULL, 'panel')
+  dt <- groupSummary(data.catXcontY, 'entity.x', 'entity.y', NULL, 'entity.panel')
   expect_is(dt, 'data.table')
   expect_equal(nrow(dt),4)
-  expect_equal(names(dt), c('panel', 'x', 'min', 'q1', 'median', 'q3', 'max'))
+  expect_equal(names(dt), c('entity.panel', 'entity.x', 'min', 'q1', 'median', 'q3', 'max'))
 
-  dt <- groupSummary(data.catXcontY, 'x', 'y', 'group', 'panel')
+  dt <- groupSummary(data.catXcontY, 'entity.x', 'entity.y', 'entity.group', 'entity.panel')
   expect_is(dt, 'data.table')
   expect_equal(nrow(dt),16)
-  expect_equal(names(dt), c('group', 'panel', 'x', 'min', 'q1', 'median', 'q3', 'max'))
+  expect_equal(names(dt), c('entity.group', 'entity.panel', 'entity.x', 'min', 'q1', 'median', 'q3', 'max'))
 })
 
 #test the more complicated ones
 test_that("groupSmoothedMean() returns an appropriately sized data.table", {
-  dt <- groupSmoothedMean(data.xy, 'x', 'y')
+
+  df <- copy(data.xy)
```

Comment:
data.table::copy

Reply:
also, why do we need to keep copying it now?

Reply:
in groupSmoothedMean and a few other group functions, we update dt names by reference. I noticed that after changing column names in data.xy to entity.varid, that when testing `groupSmoothedMean(data.xy, 'x', 'y')` would work but then the next round of `groupSmoothedMean` got a data.xy with altered column names.

Reply:
ok. then lets make the copy in groupSmoothedMean, and then update the col names on the copy.

Reply:
or update names the old school way there and get the copy for free. either way is fine w me really. but i actively decided recently that the group fxns wont work by reference